go/integration_test/processor/calls_test.go

```diff
@@ -453,6 +453,75 @@ func TestProcessor_CallingNonExistentAccountIsHandledCorrectly(t *testing.T) {
 	}
 }
 
+func TestProcessor_CallValueTransfersAreHandledCorrectly(t *testing.T) {
+	for processorName, processor := range getProcessors() {
+		for callName, call := range callTypesAndProperties() {
+			t.Run(fmt.Sprintf("%s-%s", processorName, callName), func(t *testing.T) {
+				sender0 := tosca.Address{1}
+				receiver0 := tosca.Address{2}
+				receiver1 := tosca.Address{3}
+				value := tosca.NewValue(42)
+
+				code0 := pushCallArguments(call, sufficientGas, value, receiver1)
+				code0 = append(code0, []byte{
+					byte(call.callType),
+					byte(vm.PUSH1), byte(32),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.RETURN),
+				}...)
+
+				code1 := []byte{
+					byte(vm.CALLVALUE),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.MSTORE),
+					byte(vm.PUSH1), byte(32),
+					byte(vm.PUSH1), byte(0),
+					byte(vm.RETURN),
+				}
+
+				senderBalance := tosca.NewValue(100)
+				state := WorldState{
+					sender0:   Account{Balance: senderBalance},
+					receiver0: Account{Code: code0},
+					receiver1: Account{Code: code1},
+				}
+				transaction := tosca.Transaction{
+					Sender:    sender0,
+					Recipient: &receiver0,
+					GasLimit:  sufficientGas,
+					Value:     value,
+				}
+
+				transactionContext := newScenarioContext(state)
+
+				// Run the processor
+				result, err := processor.Run(tosca.BlockParameters{}, transaction, transactionContext)
+				if err != nil || !result.Success {
+					t.Errorf("execution was not successful or failed with error %v", err)
+				}
+				if call.callType != vm.STATICCALL && !slices.Equal(result.Output, value[:]) {
+					t.Errorf("call value was not transferred correctly, got unexpected output: %v", result.Output)
+				}
+				if transactionContext.GetBalance(sender0).Cmp(tosca.Sub(senderBalance, value)) != 0 {
+					t.Errorf("sender balance was not updated correctly, got %v", transactionContext.GetBalance(sender0))
+				}
+				value0 := value
+				value1 := tosca.NewValue(0)
+				if call.callType == vm.CALL {
+					value0 = tosca.NewValue(0)
+					value1 = value
+				}
+				if transactionContext.GetBalance(receiver0).Cmp(value0) != 0 {
+					t.Errorf("receiver balance was not updated correctly, got %v", transactionContext.GetBalance(receiver0))
```

Comment:
general remark: if yo do not print the want, I would argue the got is of little value in such cases ...